datasafe-rest-impl/src/test/java/de/adorsys/datasafe/rest/impl/controller/UserControllerTest.java

```diff
@@ -43,19 +47,52 @@ void createUserTest() {
         verify(userProfile).registerUsingDefaults(any());
     }
 
+    @SneakyThrows
+    @Test
+    void createDuplicateUserTest() {
+
+        when(dataSafeService.userProfile().userExists(new UserID(TEST_USER))).thenReturn(true);
+
+        Assertions.assertThrows(NestedServletException.class, () -> {
+            mvc.perform(put("/user")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .characterEncoding("UTF-8")
+                    .content("{ \"userName\" : \"" + TEST_USER + "\" , \"password\" : \"" + TEST_PASS + "\" }")
+                    .header("token", token)
+            );
+        });
+    }
 
     @SneakyThrows
     @Test
     void deleteUserTest() {
+        if(userProfile.userExists(new UserID(TEST_USER))) {
```

Comment:
```
delete("/user")
                    .contentType(MediaType.APPLICATION_JSON)
                    .accept(MediaType.APPLICATION_JSON)
                    .header("user", TEST_USER)
                    .header("password", TEST_PASS)
                    .header("token", token)
```
Can be extracted to separate function so code won't be duplicated - it is duplicated within happy-path-and exceptional path